examples/temporal_graph_learning/gen_graph.py

```diff
@@ -0,0 +1,123 @@
+import pandas as pd
+import numpy as np
+import pyarrow.parquet as pq 
+import pyarrow as pa
+
+import torch
+import dgl
+from dgl import AddReverse, Compose, ToSimple
+
+def create_dgl_graph():
+    # load dataset
+    data = 'MAG'
+    df = pd.read_csv('./DATA/{}/edges.csv'.format(data))
+    num_nodes = max(int(df['src'].max()), int(df['dst'].max())) + 1
+    print('num_nodes: ', num_nodes)
+    
+    # Extract graph structure 
+    # We only consider edge types from 100 to 109
+    timestamp = df['time'].values
+    mask = np.logical_and(timestamp >= 100, timestamp < 110)
+    
+    timestamp = timestamp[mask]
+    src_nodes = df['src'].values[mask]
+    dst_nodes = df['dst'].values[mask]
```

Comment:
you only extracted a small number of edges?

Reply:
Yes, I first select all edges with timestamp from 100 - 110 (those are most dense graph with many labeled nodes).

Then, I only keep nodes that appears during 100 - 110, and constructed a subsampled version of MAG dataset. By doing so, we won't have nodes that doesn't have any neighbors.